packages/codegen/src/types-builder.ts

```diff
@@ -330,14 +471,28 @@ const buildSyntax = withCache(
     entry,
     declarations,
     getChecksum,
-    _knownTypes,
+    knownTypes,
     callsChecksum,
+    mode,
   ): TypeForEntry => {
     const checksum = getChecksum(entry.id)!
-    if (checksum === callsChecksum) return clientImport("TxCallData")
-    return typesImport(declarations.variables.get(checksum)!.name)
+    if (mode === Mode.Anonymize) {
+      if (checksum === callsChecksum) return clientImport("TxCallData")
```

Comment:
Moving this line above, just outside the `if` should remove multiple `__Circular` calls which is just TxCallData

Reply:
Last time I was actually thinking that expanding TxCallData would suit the docs better...
But you're probably right, hiding TxCallData makes the docs more readable.

Reply:
done